reservations/migrations/0003_from_data_to_main_fields.py

```
@@ -0,0 +1,127 @@
+        keys.split("."),
+        dictionary
+    )
+
+
+def from_data_to_main_fields(apps, schema_editor):
+    Harbor = apps.get_model('harbors', 'Harbor')
+    BoatType = apps.get_model('harbors', 'BoatType')
+    Reservation = apps.get_model('reservations', 'Reservation')
+    HarborChoice = apps.get_model('reservations', 'HarborChoice')
+
+    for r in Reservation.objects.all():
+        if r.data and r.data != {}:
+            with transaction.atomic():
+                chosen_harbors = r.data.get("selectedBerths")
+                if chosen_harbors:
+                    index = 1
+                    for harbor_identifier in chosen_harbors:
+                        harbor = Harbor.objects.filter(identifier=harbor_identifier).first()
+                        if harbor:
+                            try:
+                                HarborChoice.objects.create(
+                                    reservation=r,
+                                    harbor=harbor,
+                                    priority=index
+                                )
+                            except IntegrityError:
+                                pass
+                        index += 1
+
+                boat_type = deep_get(r.data, "boat.type")
+                if boat_type:
+                    boat_type_obj = BoatType.objects.filter(identifier=boat_type).first()
+                    if boat_type_obj:
+                        r.boat_type = boat_type_obj
+
+                for attr_name, data_address in CHAR_FIELDS_TO_DATA_KEY_MAP.items():
+                    value = deep_get(r.data, data_address)
+                    if value:
+                        setattr(r, attr_name, value)
+
+                boat_weight = deep_get(r.data, "boat.weight")
+                if boat_weight:
+                    try:
+                        r.boat_weight = int(boat_weight)
+                    except (TypeError, ValueError):
+                        pass
```

Are you sure it is safe to just silently ignore all these conversion errors, not even some warning logging needed?

---

good point! best to add at least logger 👍